library/CHART-TEMPLATE/tests/common_library_serviceaccount_name_test.yaml

```diff
@@ -44,3 +44,13 @@ tests:
       - equal:
           path: spec.template.spec.serviceAccountName
           value: default
+
+  - it: allow to override the serviceacocunt name with the local one
+    set:
+      serviceAccount:
+        name: "my-local-service-account-name"
+        create: true
+    asserts:
+      - equal:
+          path: spec.template.spec.serviceAccountName
+          value: my-local-service-account-name
```

Comment:
This test is cool and is the test that is missing but now I miss one more test.

Given these values:
```yaml
serviceAccount:
  create: false
  name: test
```
We should also test that the service account is not created and that the deployment is using "test" as a service account.

Could you take care of this in this PR to fix the `serviceAccount` once and for all?

Reply:
done